llvm/lib/CheerpUtils/Utility.cpp

```diff
@@ -741,55 +742,10 @@ bool TypeSupport::isJSExportedType(StructType* st, const Module& m)
 {
 	if (st->isLiteral())
 		return false;
-	return m.getNamedMetadata(llvm::Twine(st->getName(),"_methods"))!=NULL;
-}
-
-std::string TypeSupport::getNamespacedFunctionName(StringRef name)
-{
-	demangler_iterator demangler( name );
-
-	std::string namespacedName = "";
-
-	while (demangler != demangler_iterator())
-	{
-		namespacedName += *demangler++;
-		namespacedName += ".";
-	}
-
-	namespacedName.pop_back();
-
-	return namespacedName;
-}
-
-std::pair<StructType*, std::string> TypeSupport::getJSExportedTypeFromMetadata(StringRef name, const Module& module)
-{
-	StringRef mangledName = name.drop_back(8);;
-
-	if(name.startswith("class."))
-		mangledName = mangledName.drop_front(6);
-	else
-	{
-		assert(mangledName.startswith("struct."));
-		mangledName = mangledName.drop_front(7);
-	}
-
-
-	demangler_iterator demangler( mangledName );
-
-	std::string jsClassNameS = "";
-
-	while (demangler != demangler_iterator())
-	{
-		jsClassNameS += *demangler++;
-		jsClassNameS += ".";
-	}
-
-	jsClassNameS.pop_back();
-
-	assert( mangledName.end() > name.begin() && std::size_t(mangledName.end() - name.begin()) <= name.size() );
-	StructType * t = StructType::getTypeByName( module.getContext(), StringRef(name.begin(), mangledName.end() - name.begin() ) );
-	assert(t);
-	return std::make_pair(t, jsClassNameS);
+	for (auto record : getJsExportRecords(m))
```

Comment:
potentially quadratic since this is called for each  StructType. In practice only a handful of types will be jsexported, but still, maybe we can have a map for this lookup?